src/thor/bidirectional_astar.cc

```diff
@@ -691,18 +694,30 @@ BidirectionalAStar::GetBestPath(valhalla::Location& origin,
       } else {
         // Search is exhausted. If a connection has been found, return it
         if (best_connections_.empty()) {
-          // No route found.
-          LOG_ERROR("Bi-directional route failure - reverse search exhausted: n = " +
-                    std::to_string(edgelabels_reverse_.size()) + "," +
-                    std::to_string(edgelabels_forward_.size()));
-          return {};
+          // mark the reverse direction exhausted so we dont expand from here
+          // anymore
+          reverse_exhausted = true;
+          LOG_WARN("Reverse search exhausted: n = " + std::to_string(edgelabels_reverse_.size()) +
+                   "," + std::to_string(edgelabels_forward_.size()));
+        } else {
+          return FormPath(graphreader, options, origin, destination, forward_time_info, invariant);
         }
-        return FormPath(graphreader, options, origin, destination, forward_time_info, invariant);
       }
     }
 
-    // Expand from the search direction with lower sort cost.
-    if ((fwd_pred.sortcost() + cost_diff_) < rev_pred.sortcost()) {
+    // If both directions have exhausted, we've failed to find a route. Abort
+    if (forward_exhausted && reverse_exhausted) {
+      LOG_ERROR("Bi-directional route failure - search exhausted: n = " +
+                std::to_string(edgelabels_forward_.size()) + "," +
+                std::to_string(edgelabels_reverse_.size()));
+      return {};
+    }
+
+    // Expand from the search direction with lower sort cost
+    // Note: If one direction is exhausted, we force search in the remaining
+    // direction
```

Comment:
why do your comments seem to have a shorter line break than your code? :thinking:

Reply:
Because once upon a time (long before twitter made 120-char limits fashionable), 80-char limit was the common std for line-wrapping. I still haven't fully shrugged off that limit :smile: . I'll fix this in the next PR update.